lib/master.js

```diff
@@ -134,6 +150,10 @@ class Master extends BaseService {
             time: null,
             killed: true
         };
+
+        worker.removeListener('message', this.workerMessageHandlerMap[worker.process.pid]);
+        worker.removeListener('exit', this.workerExitHandlerMap[worker.process.pid]);
```

Comment:
This will break the restart logic. I still have to think a bit what to do here, but you have other comments to work on right now.

Reply:
Ok. Here's what we need to do. 

We need to remove the `workerExit` or `workerStartupExit` listeners from within themselves. The worker is guaranteed to exit and the 'exit' message is guaranteed to be dispatched, so it's safe. These listers have to be removed as a .finally call within themselves. Given that we can pass the references to the listener functions as a part of the closure scope, we don't need these maps. 

As for the 'message' listener - the same thing - we need to remove them within the exit listeners and we can actually pass the reference in a closure scope, so the map for that is not needed either.